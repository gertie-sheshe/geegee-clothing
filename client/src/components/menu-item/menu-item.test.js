import React from 'react';
import { shallow } from 'enzyme';

import { MenuItem } from './menu-item.component';

describe('Menu Item Tests', () => {
  let wrapper;
  let mockMatch;
  let mockHistory;

  const linkUrl = '/hats';
  const size = 'large';
  const imageUrl = 'testimage';

  beforeEach(() => {
    mockMatch = {
      url: '/shop',
    };

    mockHistory = {
      push: jest.fn(),
    };

    const mockProps = {
      match: mockMatch,
      history: mockHistory,
      linkUrl,
      size,
      title: 'hats',
      imageUrl,
    };

    wrapper = shallow(<MenuItem {...mockProps} />);
  });

  it('Should render MenuItem component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('Should call history.push with the right string when MenuItemContainer clicked', () => {
    wrapper.find('MenuItemContainer').simulate('click');
    expect(mockHistory.push).toHaveBeenCalledWith(`${mockMatch.url}${linkUrl}`);
  });

  it('Should pass size to MenuItemContainer as the prop size', () => {
    expect(wrapper.find('MenuItemContainer').prop('size')).toBe(size);
  });

  it('Should pass imageUrl to BackgroundImageContainer as the prop imageUrl', () => {
    expect(wrapper.find('BackgroundImageContainer').prop('imageUrl')).toBe(
      imageUrl,
    );
  });
});
