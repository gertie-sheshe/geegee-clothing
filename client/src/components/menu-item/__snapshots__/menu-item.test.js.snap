// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu Item Tests Should render MenuItem component 1`] = `ShallowWrapper {}`;
